org/mozilla/jss/crypto/CMACAlgorithm.java

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.jss.crypto;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Hashtable;
+
+import org.mozilla.jss.asn1.OBJECT_IDENTIFIER;
+
+/**
+ * Algorithms for performing CMACs. These can be used to create
+ * MessageDigests.
+ */
+public class CMACAlgorithm extends DigestAlgorithm {
+
+    protected CMACAlgorithm(int oidIndex, String name, OBJECT_IDENTIFIER oid,
+                            int outputSize) {
+        super(oidIndex, name, oid, outputSize);
+
+        if (oid != null && oidMap.get(oid) == null) {
+            oidMap.put(oid, this);
+        }
+    }
+
+    ///////////////////////////////////////////////////////////////////////
+    // OID mapping
+    ///////////////////////////////////////////////////////////////////////
+    private static Hashtable<OBJECT_IDENTIFIER, CMACAlgorithm> oidMap = new Hashtable<>();
+
+    /**
+     * Looks up the CMAC algorithm with the given OID.
+     *
+     * @param oid OID.
+     * @return CMAC algorithm.
+     * @exception NoSuchAlgorithmException If no registered CMAC algorithm
+     *  has the given OID.
+     */
+    public static CMACAlgorithm fromOID(OBJECT_IDENTIFIER oid)
+        throws NoSuchAlgorithmException
+    {
+        Object alg = oidMap.get(oid);
```

Comment:
It should be possible to get the `alg` as `CMACAlgorithm` directly since the `oidMap` is defined as `Hashtable<OBJECT_IDENTIFIER, CMACAlgorithm>`:
```
CMACAlgorithm alg = oidMap.get(oid);
```

Reply:
ACK, this would be a nice cleanup for DigestAlgorithm / HMACAlgorithm as well.